### DISCUSSION TITLE
How to set/get fill depth when the link type is FILLED_CIRCULAR

### DISCUSSION THREAD

**Discussion by CedarSnowy** [pyswmm/pyswmm#431] [Q&A] [answered]
When the link type is FILLED_ CIRCULAR, how should I set/get _"fill depth"_ to change the siltation situation?It seemd that "inlet_offect" is the sum of "InOffset" and _"fill depth"_  , and  "outlet_offset" is the sum of  _"fill depth"_ and "OutOffset" .

**Comment by karosc** [comment]
Unfortunately pyswmm doesn't yet support setting link dimensions, you can find all the available getters/setters in [the docs](http://wateranalytics.org/pyswmm/reference/api/pyswmm.links.Link.html). To add this functionality would require some additions to the [OWA SWMM toolkit.c file](https://github.com/OpenWaterAnalytics/Stormwater-Management-Model/blob/main/src/solver/toolkit.c#L752-L807)

**Reply by CedarSnowy** [reply]
I find it possible to read and wirte .inp file using python,maybe this can be a way to change settings

**Reply by bemcdonnell** [reply, marked as answer]
You could also check out swmmio to pre-process your INP file. That’s been a reliable and well-maintained tool!
